docs/DeveloperGuide.md

```diff
@@ -714,20 +714,158 @@ These commands should be tested on the Group Page.
 
-    1. Test case: `delete student 0`<br>
+    2. Test case: `delete student 0`<br>
        Expected: No student is deleted. Error details shown in the log box.
 
-    1. Other incorrect `delete student` commands to try: `delete`, `delete student x`, `...` (where x is larger than the list size)<br>
-       Expected: Similar to previous.
+   3. Other incorrect `delete student` commands to try: `delete`, `delete student x`, (where x is larger than the list size)<br>
+          Expected: No student is deleted. Error details shown in the log box.
 
-2. _{ more test cases …​ }_
+2. Deleting a student while the student list is empty on the Students Page.
+    1. Test case: `delete student 1`<br>
+    Expected: No student is deleted. Error details shown in the log box.
+
+#### Copying a student's email
+1. Copying a student's email to your clipboard, while there is more than one student on the Students Page.
+    1. Test case: `copy 1`<br>
+    Expected: First student's email is copied to your clipboard. Successful command message shown in the log box.
+   2. Test case: `copy 0`<br>
+   Expected: No student email is copied. Error details shown in the log box.
+   3. Other incorrect `copy` commands to try: `copy`, `copy x`, (where x is larger than the student list)
+   Expected: No student email is copied. Error details shown in the log box.
+2. Copying a student's email while the student list is empty on the Students Page.
+    1. Test case: `copy 1`<br>
+    Expected: Nothing is copied to your clipboard. Error details shown in the log box.
+
+#### Sorting students by their name or student ID
+1. Sorting student list by their name
+    1. Test case: `sort name`
+   Expected: Student list sorted by name in alphabetical order.
+2. Sorting student list by their student IDs
+    1. Test case: `sort id`
+   Expected: Student list sorted by student ID.
+
+#### Adding or deleting a remark
+1. Adding a remark to a student that does not have an existing remark.
+    1. Test case: `remark 1 Loves photography`
+   Expected: Adds a remark `Loves photography` to the first student on the student list. Details of the new remark displayed in the log box.
+   2. Test case: `remark 0 Loves sci-fi movies`
+   Expected: Displays error message on the log box.
+   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)
+   Expected: Displays error message on the log box.
+2. Adding a remark to a student that already has a remark.
+   1. Test case: `remark 1 Plays football` 
+      Expected: Replaces the previous remark that the first student has with `Plays football`. Details of the new remark displayed in the log box.
+   2. Test case: `remark 0 Loves sci-fi movies`
+     Expected: Displays error message on the log box.
+   3. Other incorrect `remark` commands to try: `remark`, `remark x`, (where x is larger than the student lit)
+      Expected: Displays error message on the log box.
+   
```

Comment:
Maybe can add cases of deleting remark, it was mentioned in the title but not listed down